### PR TITLE
Improves our suggestions for bad URLs in LoginSiteAddressViewController.

### PR DESCRIPTION
**Pull request by diegoreymendez**
## Description:

This PR improves our suggestions for bad URLs in `LoginSiteAddressViewController`, as discussed here: https://github.com/wordpress-mobile/WordPress-iOS/issues/10294

We no longer allow the user to tap "Next" unless they input a valid URL into the field (check "expectations" below for exceptions).

We also show an error message for invalid URLs after 2 seconds if there's no activity, or as soon as the user taps outside of the URL field as shown in the demo below.

## Demo

56679116-b178ee00-669a-11e9-975e-2781a5130797

## Expectations

URL validation is complex.  This PR doesn't expect to catch all invalid URLs as invalid, but at least the most obvious ones.

These changes however should not prevent you from inserting ANY valid URL into the field and clicking next.

## Testing:

You can test this PR through WPiOS's branch `issue/10294-better-error-for-invalid-url`.

1. Run WPiOS
2. At the login screen select "Login with Site Address".
3. Start playing with the URL field and make sure it works as described in this PR, and a shown in the demo above.

### CODE REVIEW

**Comment by diegoreymendez** [comment]
Sorry for assigning this earlier than I should have!  This PR is now ready for review.

**Comment by diegoreymendez** [comment]
Fixed some issues I found while doing a self-review.

This PR is a blocker for fixing the integration of `WordPress-iOS-Shared` into WPiOS.

Ping @mindgraffiti & @yaelirub in case you can do a quick review today.

**Comment by mindgraffiti** [comment]
I can't get this to build using the `issue/10294-better-error-for-invalid-url` WPiOS branch. 

There are conflicting declarations for dependencies in the WPiOS pod file in that branch. 

Screen Shot 2019-04-25 at 11 05 52 AM

File: WordPressAuthenticator/Signin/LoginSiteAddressViewController.swift [hunk 1]
```diff
@@ -305,6 +311,24 @@ class LoginSiteAddressViewController: LoginViewController, NUXKeyboardResponder
         alertController.addDefaultActionWithTitle(acceptActionTitle)
         present(alertController, animated: true)
     }
+    
+    // MARK: - URL Validation
+    
+    /// Does a local / quick Site Adrress validation and refreshes the UI with an error
```
[the remark this answers is not in the document]
Comment (diegoreymendez) [reply WordPressAuthenticator/Signin/LoginSiteAddressViewController.swift]: I somehow missed these comments.  I'll be fixing this in a new PR.

**Pull request merged**